docs/madr/decisions/073-spiffe-compliance.md

```diff
@@ -153,7 +157,711 @@ Future releases
+type: DataplaneInsight
+mTLS:
+  certificateExpirationTime: "2025-06-03T17:35:33Z"
+  certificateRegenerations: 1
+  issuedBackend: ca-2 # name of MeshIdentity
+  lastCertificateRegeneration: "2025-06-02T17:35:33.624189876Z"
+  supportedBackends: # list of MeshTrust name
+  - ca-1
+  - ca-2
+...
+```
+Update SpiffeID based on identities
+```yaml
+...
+name: second-test-server-b7dwz26bb9b4z676
+spec:
+  identities:
+  - type: SpiffeID
+    value: spiffe://example.com/ns/real-resource-ns/sa/default
+  - type: SpiffeID
+    value: spiffe://example.fr/ns/real-resource-ns/sa/default
+  ports:
+  - appProtocol: http
+    name: main
+    port: 80
+    targetPort: main
+  selector:
+    dataplaneTags:
+      app: second-test-server
+      k8s.kuma.io/namespace: real-resource-ns
+  state: Available
+status:
+  ...
+  tls:
+    status: Ready
+  vips:
+  - ip: 10.43.245.159
+type: MeshService
+```
+
+6. User can remove MeshIdentity `ca-1`
+7. MeshTrust should be removed by the user
+
+![Rotation flow](assets/073/rotation-flow.png)
+
+#### Secret naming
+
+We want to use a [KRI](070-resource-identifier.md) for resource naming. Since `MeshIdentity` creates a separate resource (Secret), it should be named based on the KRI to simplify correlation between the `Secret` and the `MeshIdentity` configuration.
+
+Proposed name:
+
+```yaml
+kri_mi_mesh-1_us-east-2_kuma-system_identity-1
+```
+
+Also, we need to name the Spire agent cluster, in this case is it internal or maybe based on MeshIdentity:
+Proposed name:
+```yaml
+system_spire-agent
+```
+
+or 
+```yaml
+kri_mi_mesh-1_us-east-2_kuma-system_spire-identity
+```
+
+> [!WARNING]
+> The naming is different when using SPIRE, since SPIRE uses SPIFFEID or `default` as a certificate name
+> https://spiffe.io/docs/latest/deploying/spire_agent/#sds-configuration
+> https://spiffe.io/docs/latest/microservices/envoy/#tls-certificates
+#### MeshTrust
+
+Based on the existing `MeshIdentity`, we should create a `MeshTrust` resource, which will be used to properly generate trust domains for other zones.
+`MeshTrust` should be created in each zone and then synchronized with the global control plane.
+1. Can be created on the Zone and synced to Global
+2. Can be created on the Global and synced to all Zones
+3. Can be created on the Zone and not synced to global by providing label `kuma.io/kds-sync: disabled`
+4. Can be created on the Zone and synced to all zones (Future release) (`SyncedAcrossZonesFlag`)
+
+We can describe it as KDS flag:
+
+```golang
+model.GlobalToZonesFlag | model.ZoneToGlobalFlag
+```
+
+There should also be an option to disable automatic generation in cases where the user does not want cross trust-domain traffic to be allowed by default.
+
+**Model**
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: MeshTrust
+metadata:
+  name: trust-hash
+  namespace: kuma-system # only in system namespace
+  labels:
+    kuma.io/mesh: default
+spec:
+  ca: inline | secret | file
```

Comment:
That's a caBundle right?

Reply:
Btw the CA rotation without trustDomain change is it just a matter of having the old and new bundle and then removing the old once it's expired or if a user wants?

Reply:
would it be useful to have an optional origin?